### DISCUSSION TITLE
NEWS FOR 2024-03-31

### DISCUSSION THREAD

**Discussion by samiam95124** [samiam95124/Pascal-P6#132] [Announcements]
************************************************ NEWS FOR 2024-02-25 *************************************************

* Pascal-P6 bootstraps itself!
* Now possible to run Pascal-P6 without GPC or other host.
* New host configuration, iso7185, available.

pgen is now able to compile the entire Pascal-P6 system. The dependency now is gcc/gas (the C compiler and its assembler). Whatever the latest gcc is available will do, it is not dependent on a particular version. It should also work with llvm, but I haven't created a configuration for this as yet.

I have committed a new host configuration for the self compile. It is:

./configure --iso7185

FAQs over the new system:

1. It compiles iso7185 mode code only. It will *interpret* Pascaline code as before. If you have been following along, you will know that the compiler and interpreter is written in iso7185 Pascal only.
2. Testing is early stage. It runs only partial regressions.
3. It is unoptomized. Although speed as I have tested, is close or equal to other compilers (GPC), this is early days and it will get much better.
4. It is only tested on Ubuntu 20.04.6 LTS (my machine). I can't vouch for other Linux versions and types.
5. It can't yet interface with external libraries, that is dependent on Pascaline implementation.

In short, this is a big advance for Pascal-P6. More will follow.

**Comment by Hyrtwol** [comment]
So free at last :) Congratulations with you'r progress on P6! Crazy commit rate. I just installed Ubuntu on my new windows 10 :p so will give it a go.

**Comment by samiam95124** [comment]
Great to hear from a user!

Keep me informed if you have any issues.

Scott Franco
